dali/operators/generic/lookup_table.cu

```diff
@@ -24,57 +25,67 @@ namespace detail {
 constexpr auto kMaxKey = LookupTable<GPUBackend>::kMaxKey;
 
 template <typename OutputType, typename InputType>
-__global__ void
-LookupValuesImpl(OutputType *output,
-                 const InputType *input,
-                 size_t data_size,
-                 const OutputType* lookup_table,
-                 const OutputType default_value) {
+__global__ void LookupValuesImpl(const LutSampleDesc *samples, const kernels::BlockDesc<1> *blocks,
+                                 const OutputType *lookup_table, const OutputType default_value) {
   // We do not check the key range when the type range is smaller than the supported range
   constexpr bool check_range =
-       !std::is_same<InputType, uint8_t>::value
-    && !std::is_same<InputType, uint16_t>::value;
+      !std::is_same<InputType, uint8_t>::value && !std::is_same<InputType, uint16_t>::value;
   constexpr auto max_key = ConvertSat<InputType>(kMaxKey);
 
-  size_t tid = blockDim.x * blockIdx.x + threadIdx.x;
-  if (tid >= data_size)
-    return;
+  const auto &block = blocks[blockIdx.x];
+  const auto &sample = samples[block.sample_idx];
 
-  const auto key = input[tid];
-  if (check_range) {
-    output[tid] = (std::is_unsigned<InputType>::value || key >= 0) && key <= max_key ?
-      lookup_table[key] : default_value;
-  } else {
-    output[tid] = lookup_table[key];
+  auto *output = reinterpret_cast<OutputType *>(sample.output);
+  const auto *input = reinterpret_cast<const InputType *>(sample.input);
+  for (int x = threadIdx.x + block.start.x; x < block.end.x; x += blockDim.x) {
```

Comment:
I have some reservations for the use of BlockSetup here. It was initially meant to work with multidimensional data and does not support tensors in excess of 2G in each dimension - not a problem for large spatial problems, but for a flattened tensor we could hit this limit pretty easily and face dire consequences.

Reply:
Addressed here: https://github.com/NVIDIA/DALI/pull/3304

Reply:
```suggestion
  for (auto x = threadIdx.x + block.start.x; x < block.end.x; x += blockDim.x) {
```